File: eCakeShop.Services/Database/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCakeShop.Services.Database
{
    public class Korisnik
    {
        public Korisnik()
        {
            KorisnikUlogas = new HashSet<KorisnikUloga>();
        }
        public int KorisnikID { get; set; }
        public string Ime { get; set; } = null!;
        public string Prezime { get; set; } = null!;
        public DateTime DatumRodjenja { get; set; }
        public string? Email { get; set; }
        public string? Telefon { get; set; }
        public int GradID { get; set; }
        public int DrzavaID { get; set; }
        public string KorisnickoIme { get; set; } = null!;
        public string LozinkaHash { get; set; } = null!;
        public string LozinkaSalt { get; set; } = null!;

        public virtual Drzava Drzava { get; set; } = null!;
        public virtual Grad Grad { get; set; } = null!;
        public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; }
    }
}

File: UI/e_cakeshop_admin/lib/models/korisnik.dart
import 'package:e_cakeshop_admin/models/drzava.dart';
import 'package:e_cakeshop_admin/models/grad.dart';
import 'package:e_cakeshop_admin/models/uloga.dart';
import 'package:json_annotation/json_annotation.dart';
part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int? korisnikID;
  String? ime;
  String? prezime;
  String? korisnickoIme;
  DateTime? datumRodjenja;
  String? email;
  String? telefon;
  String? lokacija;
  int? gradID;
  int? drzavaID;
  Grad? grad;
  Drzava? drzava;
  String? lozinka;
  Uloga? uloga;
  String? uloge;

  Korisnik({
    this.ime,
    this.prezime,
    this.korisnickoIme,
    this.datumRodjenja,
    this.email,
    this.telefon,
    this.lokacija,
    this.gradID,
    this.drzavaID,
    this.grad,
    this.drzava,
    this.lozinka,
    this.uloga,
    this.uloge,
  });

  factory Korisnik.fromJson(Map<String, dynamic> json) =>
      _$KorisnikFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikToJson(this);
}
